src/main/java/com/google/api/codegen/transformer/SurfaceNamer.java

```diff
@@ -1027,4 +1027,12 @@ public String getApiCallableTypeName(ServiceMethodType serviceMethodType) {
   public String getReleaseAnnotation(ReleaseLevel releaseLevel) {
     return getNotImplementedString("SurfaceNamer.getReleaseAnnotation");
   }
+
+  public String getMetadataIdentifier() {
+    return getNotImplementedString("SurfaceNamer.getMetadataIdentifier");
+  }
+
+  public String getMetadataName() {
```

Comment:
Yes, the metadata names are conceptually different from the repo language. I will do a separate class.